api-gateway-routes/api_gateway_proxy_function.rb

```diff
@@ -27,7 +27,15 @@ def on_connect(event, context)
   authorizer = request_context['authorizer']
 
   authorization_error_response = AuthErrorResponseHelper.get_response(authorizer)
-  return authorization_error_response unless authorization_error_response == nil
+  # If there is an authorization error, allow the websocket connection to open,
+  # but early return so we don't actually invoke the Build and Run lambda and try
+  # to create an SQS queue. We will send the auth error message via the websocket
+  # once the connection has been successfully made (in on_default).
```

Comment:
clarifying q: is the reason the browser doesn't interpret the 403 because we haven't implemented that on the java lab side or because it's not possible to interpret it from the browser side?

Reply:
So from what I understand, it's because the browser just doesn't interpret it/handle it. A 403 here would result in the handshake portion of the websocket connection failing, which on the browser just appears as a websocket error.

Reply:
Does the browser differentiate between different error codes? i.e. I could see us returning a 403 for this case, but a different error code for generic errors.

Reply:
Hmm, not that I can tell for this specific part of the WebSocket connection. I believe if it's anything other than 200 here, the socket will just close with an error. For reference: https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers#the_websocket_handshake

That being said, we can set custom error codes for auth errors in our HTTP API. For now, I just went off the throttling doc and always set it to 403 ([here](https://github.com/code-dot-org/javabuilder/blob/c32ddb81a07ccae586cb1b62cb6d8b5bc6dd4d38/javabuilder-authorizer/jwt_helper.rb#L47)), but if it makes sense to have different error codes for different auth situations we can definitely support that.